scripts/mole/mole.gd

```diff
@@ -14,15 +14,26 @@ signal mole_died(data: MoleData)
 
 
 var max_health: int:
-	get: return _max_health
+	get: return max_health
```

Comment:
Does this create a public getter with private setter?

Reply:
I don't believe there is a real difference between private and public variables or functions in Godot - prefixing something with an underscore means that it isn't visible in the public documentation. Hence, by having both a private (`_max_health`) and a public variable (`max_health`) intended to store the same data, you're not actually getting the benefit of restricting access to the private variable, you're just duplicating code and data.

![image](https://github.com/tessapower/week-sauce-feb24/assets/25911223/642f3626-34a7-4092-b754-aa69b0d0b451)

These getters and setters are generally used to wrap read/write access and do other things around simply setting the value, e.g. clamping it to a range. Every property is always accessible through dot notation, and this is a bit of syntactic sugar that allows you to go `Obj.get("some_property")`, when `Object.some_property` will achieve the same thing.

Reply:
Ah okay